manticore/platforms/linux.py

```diff
@@ -1228,43 +1244,40 @@ def _open(self, f) -> int:
             self.files.append(f)
         return fd
 
-    def _close(self, fd: int) -> None:
+    def _close(self, fd):
         """
         Removes a file descriptor from the file descriptor list
         :rtype: int
         :param fd: the file descriptor to close.
-        :return: C{0} on success.
         """
         try:
-            self.files[fd].close()
-            self._closed_files.append(
-                self.files[fd]
-            )  # Keep track for SymbolicFile testcase generation
+            f = self.files[fd]
+            if f is None:
+                raise FdError(f"Bad file descriptor ({fd})", errno.EBADF)
+            f.close()
+            self._closed_files.append(f)  # Keep track for SymbolicFile testcase generation
             self.files[fd] = None
         except IndexError:
             raise FdError(f"Bad file descriptor ({fd})", errno.EBADF)
 
-    def _dup(self, fd: int) -> int:
-        """
-        Duplicates a file descriptor
-        :rtype: int
-        :param fd: the file descriptor to duplicate.
-        :return: C{0} on success.
-        """
-        return self._open(self.files[fd])
-
```

Comment:
Note: I got rid of this because it was used in just a single place; simpler to inline it and delete code.